### PR TITLE
Prac 6 code review request

### PR DESCRIPTION
**Pull request by JJaace**
@1428-wsx-f1

### CODE REVIEW

File: prac_06/guitar.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+class Guitar:
```
Comment (1428-wsx-f1) [inline prac_06/guitar.py:1]: Missing docstring for class

File: prac_06/guitar.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    def __init__(self, name="", year=0, cost=0):
```
Comment (1428-wsx-f1) [inline prac_06/guitar.py:4]: Missing docstrings for functions

File: prac_06/guitar.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+        self.name = name
+        self.year = year
+        self.cost = cost
+
+    def get_age(self):
+        current_year = 2023
```
Comment (1428-wsx-f1) [inline prac_06/guitar.py]: current_year would be better off as a constant

File: prac_06/guitar.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+        age = current_year - self.year
```
Comment (1428-wsx-f1) [inline prac_06/guitar.py]: Instead of defining a variable here you can just return current_year - self.year

File: prac_06/guitar.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+        return age
+
+    def is_vintage(self):
+        return self.get_age() >= 50
```
Comment (1428-wsx-f1) [inline prac_06/guitar.py]: The 50 looks a bit like a magic number, replace it with a constant

File: prac_06/guitars.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+            cost = float(input("Cost: $"))
+        except ValueError:
+            print("Invalid input. Please enter a valid year and cost.")
+            name = input("Name: ")
+            continue
+
+        if year < 0 or cost < 0:
+            print("Year and cost must be non-negative.")
+            name = input("Name: ")
+            continue
```
Comment (1428-wsx-f1) [inline prac_06/guitars.py]: You could also create a reusable function by passing  a prompt as parameter

File: prac_06/guitars.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+        guitar = Guitar(name, year, cost)
+        guitars.append(guitar)
+        print(guitar)
+        guitar_number += 1
```
Comment (1428-wsx-f1) [inline prac_06/guitars.py]: guitar_number doesn't do anything, might as well remove

File: prac_06/programming_language.py [hunk 8]
```diff
@@ -0,0 +1,12 @@
+class ProgrammingLanguage:
+    def __init__(self, name="Language", typing="Type", reflection=False, appeared=0):
```
Comment (1428-wsx-f1) [inline prac_06/programming_language.py]: appeared is confusing, year would be better

File: prac_06/guitars.py [hunk 9]
```diff
@@ -0,0 +1,44 @@
+from prac_06.guitar import Guitar
```
Comment (1428-wsx-f1) [inline prac_06/guitars.py:5]: Missing estimated time and completion time

**Review by 1428-wsx-f1 — COMMENTED** [review]
Good job, just a few things to fix up

**Comment by lindsaymarkward** [comment]
@JJaace - don't forget to use words in your request. Follow the guidelines for asking for something.

**Pull request merged**